README.md

```diff
@@ -1 +1,322 @@
 # AbsaOSS/k3d-action
+Easy to use GitHub action allows you to test single/multi-cluster apps with additional private registry support.
+
+- [Introduction](#introduction)
+- [Getting started](#getting-started)
+- [Single Cluster](#single-cluster)
+- [Multi Cluster](#multi-cluster)
+    - [Multi Clusters on default network](#multi-cluster-on-default-network)
+    - [Multi Cluster on isolated networks](#multi-cluster-on-isolated-networks)
+    - [Two pairs of clusters on two isolated networks](#two-pairs-of-clusters-on-two-isolated-networks)
+- [Private Registry](#private-registry)
+    - [Single Cluster](#single-cluster-with-private-registry)
+    - [Multi Cluster](#multi-cluster-with-private-registry)
+    - [Custom Private Registry](#using-custom-private-registry)
+
+## Introduction
+
+Applications running on kubernetes clusters (microservices, controllers,...) come with their own set of complexities and concerns. 
+In particular, E2E testing k8s based applications requires new approaches to confirm proper operation and continued 
+availability under heavy load or in the face of resource failure. **AbsaOSS/k3d-action allows to test the** _overall_ **application 
+functionality**. For instance, the E2E use-case is [operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) testing
+ in [AbsaOSS/k8gb](https://github.com/AbsaOSS/k8gb). 
+
+## Getting started
+AbsaOSS/k3d-action runs [k3d](https://k3d.io/) which is a lightweight wrapper to run [k3s](https://k3s.io/) 
+(Rancher Lab’s minimal Kubernetes distribution) in a docker. Thanks to that, we could spin up the test environment 
+quickly with minimal memory requirements, which is especially important in multi-cluster environments.
+
+AbsaOSS/k3d-action defines several input attributes and two outputs:
+
+### Inputs
+ - `cluster-name` (Required) Cluster name.
+ 
+ - `args` (Optional) list of k3d arguments defined by [k3d command tree](https://k3d.io/usage/commands/)
+ 
+ - `network` (Optional) Cluster network name. AbsaOSS/k3d-action primarily creates clusters in the default bridge-network 
+ called  `k3d-action-bridge-network` with subnet CIDR `172.16.0.0/24`. You can leave this field empty until you  need to 
+ have a multiple clusters in different subnets. 
+ 
+ - `subnet-CIDR` (Optional) Cluster subnet CIDR. Provide new CIDR only if `network` is defined first time.
+ 
+ - `use-default-registry` (Optional) If `true`, [injects](https://github.com/AbsaOSS/k3d-action/runs/1537165520?check_suite_focus=true#step:3:34) 
```

Comment:
We probably shouldn't include build log links into the documentation.
Logs will expire after 90 days, see
https://docs.github.com/en/free-pro-team@latest/actions/reference/usage-limits-billing-and-administration#artifact-and-log-retention-policy

Reply:
thx, will fix and amend changed